hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -39,11 +40,35 @@ import Hydra.Ledger.Cardano.Builder (
  )
 import Hydra.Party (Party, partyFromChain, partyToChain)
 import Hydra.Snapshot (Snapshot (..), SnapshotNumber)
-import Plutus.V1.Ledger.Api (fromBuiltin, fromData, toBuiltin)
+import Plutus.V1.Ledger.Api (POSIXTime, fromBuiltin, fromData, toBuiltin)
 import qualified Plutus.V1.Ledger.Api as Plutus
 
+-- | Needed on-chain data to create Head transactions.
 type UTxOWithScript = (TxIn, TxOut CtxUTxO, ScriptData)
 
+-- | Representation of the Head output after an Init transaction.
+data InitialThreadOutput = InitialThreadOutput
+  { initialThreadUTxO :: UTxOWithScript
+  , initialContestationPeriod :: OnChain.ContestationPeriod
+  , initialParties :: [OnChain.Party]
+  }
+  deriving (Eq, Show)
```

Comment:
I like the direction in which this is going. As I said it yesterday, this is very similar to the `InitObservation` and I'm quite keen to unify them :)